tests/bsim/bluetooth/audio/src/bap_scan_delegator_test.c

```diff
@@ -708,13 +712,49 @@ static void remove_all_sources(void)
 
 			printk("[%zu]: Source removed with id %u\n",
 			       i, state->src_id);
+		}
+	}
+}
 
-			printk("Terminating PA sync\n");
-			err = pa_sync_term(state);
-			if (err) {
-				FAIL("[%zu]: PA sync term failed (err %d)\n", err);
-				return;
-			}
+static void terminate_all_pa(void)
+{
+	int err;
+
+	for (size_t i = 0U; i < ARRAY_SIZE(sync_states); i++) {
+		struct sync_state *state = &sync_states[i];
+
```

Comment:
```suggestion

	for (size_t i = 0U; i < ARRAY_SIZE(sync_states); i++) {
		struct sync_state *state = &sync_states[i];
		int err;

```

Minor non-blocking suggestion